test/e2e/helpers.go

```diff
@@ -684,22 +684,28 @@ func resolveKubetestRepoListPath(version string, path string) (string, error) {
 	}
 
 	if version == "latest" {
-		return filepath.Join(path, "repo-list.yaml"), nil
+		return "", nil
 	}
 
+	// - prior to K8s v1.21 repo-list-k8sprow.yaml should be used
+	//   since all test images need to come from k8sprow.azurecr.io
+	// - starting with K8s v1.25 repo lists repo list is not needed
+	// - use repo-list.yaml for everything in between which has only
+	//   some images in k8sprow.azurecr.io
```

Comment:
nit: move this to line 705/711 where `repo-list.yaml` is referenced: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2337/files#diff-ead556dbd7c8058f584b092a1158049cf3350fc235fdeb762b2a4486abb610c4R711